tests/test_interface.py

```diff
@@ -399,6 +415,28 @@ def test_interface_delete_payment_with_invalid_id(self, mock_db):
         self.assertNotEqual(payment_after_deletion.id,invalid_id)
         #TODO - next sprint : do get, check count, should be same
 
+#valid PUT data
+    @mock.patch.object(PaymentService,'is_valid_put',return_value = True)
+    @mock.patch.object(app_db, 'session')
+    @mock.patch('app.db.models.Payment')
+    #@mock.patch.object(Payment, 'deserialize_put')
+    #@mock.patch.object(Payment, 'serialize', return_value=PUT_CREDIT_RETURN)
+    def test_interface_update_with_valid_put_data(self,mock_isvalid,mock_db,mock_P):
+        print 'in here'
```

Comment:
Yeah, I think Derek is on the right track here.  The one time that you'd really have to worry about doing the "return value mock" of a call with specific arguments (i.e., `.get(1).return_value`) is when the function call has some unique kind of chaining.  What I mean here is a function call where it only occurs once and in a very particular way.  For instance, maybe you would need to mock the return value of `get(1).deserialize().to_json()`; in that case, you'd need to supply that specific argument for the `get`.  Otherwise, if the `get` the last chained method call, just deal with the return value and don't worry about the arguments passed.